File: user/src/main/scala/ru/tdd/app/models/dto/BaseDTO.scala
package ru.tdd.app.models.dto

import zio.json.{DeriveJsonDecoder, DeriveJsonEncoder}

/**
 * @author Tribushko Danil
 * @since 20.11.2025
 *        Dto класс объекта с одним полем-названием
 */
case class BaseDTO(name: String)

object BaseDTO {

  implicit val encoder = DeriveJsonEncoder.gen[BaseDTO]
  implicit val decoder = DeriveJsonDecoder.gen[BaseDTO]
}

File: telegram-bot/src/main/java/ru/tdd/telegram_bot/model/dto/BaseDTO.java
package ru.tdd.telegram_bot.model.dto;

/**
 * @author Tribushko Danil
 * @since 20.12.2025
 * Dto объекта имеющего одно поле
 */
public class BaseDTO {

    private String name;

    public BaseDTO() {}

    public BaseDTO(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }
}
